scripts/choco-install.ps1

```diff
@@ -0,0 +1,25 @@
+# Add the cached packages with source priority 1 (Chocolatey community is 0)
+choco source add --name="nodejs" --source=".\tmp\chocolatey\nodejs.install" --priority=1
+choco source add --name="python" --source=".\tmp\chocolatey\python3" --priority=1
+
+# Install nodejs v16.14.2 (will use cahe if exists)
+choco install nodejs.install --version="16.14.2" --require-checksums -y
+# Internalise nodejs to cache
+Rename-Item -Path "C:\ProgramData\chocolatey\lib\nodejs.install\nodejs.install.nupkg" -NewName "nodejs.install.nupkg.zip" -ErrorAction:SilentlyContinue
+Expand-Archive -LiteralPath "C:\ProgramData\chocolatey\lib\nodejs.install\nodejs.install.nupkg.zip" -DestinationPath "C:\ProgramData\chocolatey\lib\nodejs.install" -Force
+Remove-Item "C:\ProgramData\chocolatey\lib\nodejs.install\_rels" -Recurse
+Remove-Item "C:\ProgramData\chocolatey\lib\nodejs.install\package" -Recurse
+Remove-Item "C:\ProgramData\chocolatey\lib\nodejs.install\[Content_Types].xml"
+New-Item -Path ".\tmp\chocolatey\nodejs.install" -ItemType "directory" -ErrorAction:SilentlyContinue
+choco pack "C:\ProgramData\chocolatey\lib\nodejs.install\nodejs.install.nuspec" --outdir ".\tmp\chocolatey\nodejs.install"
```

Comment:
Instead of specifying `C:\ProgramData\chocolatey`, you can use 

```
PS C:\Users\Roger Qiu> echo $env:ChocolateyInstall
C:\ProgramData\chocolatey
```

So you can replace it with `"${env:ChocolateyInstall}\lib\nodejs.install"`.

This env variable should always be available.

Also I suggest putting `nodejs.install` as a local variable like `$package = "nodejs.install` then reusing it these commands you have above.

This also internalises it repeatedly. My suggestion to catch an exception only applied if there was a single command to deal with it. If there's no single command, use your if condition that you had before where you checked whether the directory already existed.

We can always "wipe" the cache directly on gitlab CI/CD where we can delete cache on the panel.